Source/mhd/mhd_eigen.H

```diff
@@ -0,0 +1,622 @@
+#ifndef CASTRO_MHD_EIGEN_H
+#define CASTRO_MHD_EIGEN_H
+
+#include "mhd_util.H"
+
+void
+AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+Castro::evals(Real* lam, const Real as_in, Real* Q, const int dir) {
```

Comment:
Can we use Array1D for things like `lam` and `Q` rather than raw pointers?

Reply:
done